### PR TITLE
Initial version of OpenBSD docs

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
Overview, VM automation, kernel customization and debugging.

### CODE REVIEW

File: openbsd/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+A custom package for OpenBSD is prepared and is picked up by installation which
+makes system useable on the first run.
+
+Resources:
+ * https://man.openbsd.org/autoinstall
+ * https://man.openbsd.org/install.site.5
+ * https://www.skreutz.com/posts/autoinstall-openbsd-on-qemu/
+ * https://eradman.com/posts/autoinstall-openbsd.html
+
+## Running OpenBSD in QEMU manually
```
Comment (tomzy-0) [inline openbsd/README.md:49]: @SergiiDmytruk it would be nice to add here info how to get `OVMF.fd`

File: openbsd/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+# for installation
+qemu-system-x86_64 -m 2048 \
+                   -hda miniroot71.img \
+                   -drive if=virtio,file=obsd.qcow2,format=qcow2 \
+                   -enable-kvm \
+                   -netdev user,id=mynet0 \
+                   -device virtio-net,netdev=mynet0 \
+                   -bios OVMF.fd \
+                   -smp 6 \
```
Comment (tomzy-0) [inline openbsd/README.md]: I would add `-smp $(nproc)` to not overstate the host capability

File: openbsd/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                   -cpu host
+
+# for use (including debugging)
+qemu-system-x86_64 -m 2048 \
+                   -drive if=virtio,file=obsd.qcow2,format=qcow2 \
+                   -enable-kvm \
+                   -netdev user,id=mynet0,hostfwd=tcp:127.0.0.1:9272-:22 \
+                   -device virtio-net,netdev=mynet0 \
+                   -bios OVMF.fd \
+                   -smp 6 \
```
Comment (tomzy-0) [inline openbsd/README.md]: Like above

File: openbsd/README.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+config CUSTOM
+
+# build the kernel for the first time (takes about 3 minutes)
+make -C ../compile/CUSTOM -j6
+
+# copy the kernel image into / for convenience
+doas cp /sys/arch/amd64/compile/CUSTOM/obj/bsd /bsd.custom
+```
+
+Now can reboot and run `> bios bsd.custom` command in bootloader to load newly
```
Comment (tomzy-0) [inline openbsd/README.md]: This is not working for me. Should not we run `set image bsd.custom` then `boot` in bootloader?

File: openbsd/vm/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+`bash`, `vim` and `git` are installed.
+
+X11 packages are also installed in case of port development, because ports
+refuse to build if these are missing.
+
+SSH server is running and forwarded on port 9272 on the host.
+
+`root` has password `root` and it can't log in via SSH.
+
+`user` has no password, but it will have your `~/.ssh/id_rsa.pub` listed in
```
Comment (tomzy-0) [inline openbsd/vm/README.md:42]: Should console login (not via SSH) should be accessible for that user? I tried but get password prompt so was not able to.

File: openbsd/vm/run.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    exit 1
+fi
+
+qemu-system-x86_64 -m 2048 \
+                   -drive if=virtio,file=disk.qcow2,format=qcow2 \
+                   -enable-kvm \
+                   -netdev user,id=mynet0,hostfwd=tcp:127.0.0.1:9272-:22 \
+                   -device virtio-net,netdev=mynet0 \
+                   -bios OVMF.fd \
+                   -smp 6 \
```
Comment (tomzy-0) [inline openbsd/vm/run.sh]: Maybe we could set here $(nproc)?

File: openbsd/vm/setup.sh [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+    qemu-img create -f qcow2 disk.qcow2 32G
+
+    qemu-system-x86_64 -m 2048 \
+                       -drive if=virtio,file=disk.qcow2,format=qcow2 \
+                       -enable-kvm \
+                       -netdev user,id=mynet0,tftp=tftp/,bootfile=auto_install \
+                       -device virtio-net,netdev=mynet0 \
+                       -bios OVMF.fd \
+                       -smp 6 \
```
Comment (tomzy-0) [inline openbsd/vm/setup.sh]: Maybe we could set here $(nproc)?

**Review by tomzy-0 — CHANGES_REQUESTED** [review]
@SergiiDmytruk I leave some review, could you please address it?

File: openbsd/README.md (hunk 4 above)
Comment (SergiiDmytruk) [reply openbsd/README.md]: `bsd.custom` should work as well, at least as a first command.

File: openbsd/vm/README.md (hunk 5 above)
Comment (SergiiDmytruk) [reply openbsd/vm/README.md:42]: No, as there is no password. Can log in as root and do `su user`.

File: openbsd/README.md (hunk 4 above)
Comment (tomzy-0) [reply openbsd/README.md]: `> bios` at the beginning mislead me, just `bsd.custom` works fine

**Review by tomzy-0 — APPROVED** [review]

**Pull request merged**